src/table/table.scss

```diff
@@ -196,12 +196,12 @@
         }
       }
 
-      > .iui-slot > .iui-more-options {
+      > .iui-slot:not(.iui-disabled) > .iui-more-options {
```

Comment:
I started thinking to remove `iui-more-options` class because we are not using it in React, nor I really know how to expose it. The only idea is to have `TableMoreOptions` component that is just a wrapper on dropdown button and only adds this class.